### ISSUE TITLE
AVA_ENALBE_DEBUG doesn't work

### ISSUE THREAD

**Issue by vancemiller** [utcs-scea/ava#78]
```
# in build directory
cmake -DAVA_ENABLE_DEBUG .
make
```
results in:
```
...
CMakeFiles/worker.dir/home/vance/Documents/workspace/ava/common/cmd_channel_socket_tcp.cpp.o: In function `command_channel_socket_tcp_guest_new':
/home/vance/Documents/workspace/ava/common/cmd_channel_socket_tcp.cpp:50: undefined reference to `guestconfig::config'
/home/vance/Documents/workspace/ava/common/cmd_channel_socket_tcp.cpp:56: undefined reference to `guestconfig::config'
/home/vance/Documents/workspace/ava/common/cmd_channel_socket_tcp.cpp:57: undefined reference to `guestconfig::config'
/home/vance/Documents/workspace/ava/common/cmd_channel_socket_tcp.cpp:126: undefined reference to `guestconfig::config'
collect2: error: ld returned 1 exit status
CMakeFiles/worker.dir/build.make:328: recipe for target 'worker' failed
make[8]: *** [worker] Error 1
CMakeFiles/Makefile2:123: recipe for target 'CMakeFiles/worker.dir/all' failed
make[7]: *** [CMakeFiles/worker.dir/all] Error 2
Makefile:148: recipe for target 'all' failed
make[6]: *** [all] Error 2
CMakeFiles/cudadrv-nw.dir/build.make:130: recipe for target 'cudadrv_nw/src/cudadrv-nw-stamp/cudadrv-nw-build' failed
make[5]: *** [cudadrv_nw/src/cudadrv-nw-stamp/cudadrv-nw-build] Error 2
CMakeFiles/Makefile2:123: recipe for target 'CMakeFiles/cudadrv-nw.dir/all' failed
make[4]: *** [CMakeFiles/cudadrv-nw.dir/all] Error 2
Makefile:102: recipe for target 'all' failed
make[3]: *** [all] Error 2
CMakeFiles/ava-spec.dir/build.make:130: recipe for target 'ava-spec/src/ava-spec-stamp/ava-spec-build' failed
make[2]: *** [ava-spec/src/ava-spec-stamp/ava-spec-build] Error 2
CMakeFiles/Makefile2:117: recipe for target 'CMakeFiles/ava-spec.dir/all' failed
make[1]: *** [CMakeFiles/ava-spec.dir/all] Error 2
Makefile:102: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by vancemiller** [comment]
The built-in CMAKE_BUILD_TYPE can accomplish the same thing. I tried removing AVA_BUILD_TYPE but ended up with the same error. Something about the debug build is broken?

**Comment by vancemiller** [comment]
The problem here seems to be caused by changing the optimization level to `-O0`, from `-O2` resulting in different handling for an undefined varibale. If you change `cava/nightwatch/generator/c/cmakelists.py:28` from `-O2` to `-O0` and leave `AVA_ENABLE_DEBUG` unset the same compilation error occurs.

Is `std::shared_ptr<GuestConfig> config;` expected to be valid in the worker? I only see the variable definition in `guestlib/src/guest_config.cpp` and this is compiled into the guestlib. The worker is not linked against the guestlib so this symbol is undefined. Seems like this should either be removed from the worker or compiled in a separate library which is linked to the worker and the guestlib.

**Comment by yuhc** [comment]
Good catch! It should be fixed by PR #80.
